### PR TITLE
Tune configs

### PR DESCRIPTION
**Pull request by thabarbados**
tune project configs
remove debounce dependency and add utils with the same code

### CODE REVIEW

File: src/lib/debounce.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+interface DebouncedPromise<FunctionReturn> {
+  resolve: (result: FunctionReturn) => void;
+  reject: (reason?: any) => void;
+}
+
+export function debounce<Args extends any[], F extends (...args: Args) => any>(
+  func: F,
+  waitMilliseconds = 50,
+  options: Options<ReturnType<F>> = {}
```
Comment (dangreen) [inline src/lib/debounce.ts:25]: currently we dont use debounce with options - lets remove it to reduce size?

File: src/lib/debounce.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      if (shouldCallNow) {
+        const result = func.apply(context, args);
+        callback && callback(result);
+        return resolve(result);
+      }
+      promises.push({ resolve, reject });
+    });
+  };
+
+  debouncedFunction.cancel = function (reason?: any) {
```
Comment (dangreen) [inline src/lib/debounce.ts:81]: also this is not used

**Review by dangreen — CHANGES_REQUESTED** [review]

File: src/lib/debounce.ts (hunk 1 above)
Comment (thabarbados) [reply src/lib/debounce.ts]: removed

File: src/lib/debounce.ts (hunk 2 above)
Comment (thabarbados) [reply src/lib/debounce.ts]: removed

File: src/lib/debounce.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  function nextInvokeTimeout() {
+    return waitMilliseconds;
+  }
+
+  const debouncedFunction = function (
+    this: ThisParameterType<F>,
+    ...args: Parameters<F>
+  ) {
+    const context = this;
+    return new Promise<ReturnType<F>>((resolve, reject) => {
```
Comment (dangreen) [inline src/lib/debounce.ts:32]: also we dont use this

Comment (thabarbados) [reply src/lib/debounce.ts]: changed

**Review by dangreen — APPROVED** [review]

**Pull request merged**